examples/step-81/step-81.cc

```diff
@@ -0,0 +1,799 @@
+    result[1][0] = point[0] * point[1] * (d_1 - d_2);
+    result[1][1] = point[1] * point[1] * d_1 + point[0] * point[0] * d_2;
+    return result;
+  }
+
+
+  template <int dim>
+  typename PerfectlyMatchedLayer<dim>::rank2_type
+  PerfectlyMatchedLayer<dim>::a_matrix(const Point<dim, double> point)
+  {
+    const auto d     = d_tensor(point);
+    const auto d_bar = d_bar_tensor(point);
+    return invert(T_exer(d * d, d * d_bar, point)) *
+           T_exer(d * d, d * d_bar, point);
+  }
+
+
+  template <int dim>
+  typename PerfectlyMatchedLayer<dim>::rank2_type
+  PerfectlyMatchedLayer<dim>::b_matrix(const Point<dim, double> point)
+  {
+    const auto d     = d_tensor(point);
+    const auto d_bar = d_bar_tensor(point);
+    return invert(T_exer(d, d_bar, point)) * T_exer(d, d_bar, point);
+  }
+
+
+  template <int dim>
+  typename PerfectlyMatchedLayer<dim>::rank2_type
+  PerfectlyMatchedLayer<dim>::c_matrix(const Point<dim, double> point)
+  {
+    const auto d     = d_tensor(point);
+    const auto d_bar = d_bar_tensor(point);
+    return invert(T_exer(1. / d_bar, 1. / d, point)) *
+           T_exer(1. / d_bar, 1. / d, point);
+  }
+
+
+  // @sect4{Maxwell Class}
+  // At this point we are ready to instantiate all the major functions of
+  // the finite element program and also a list of variables.
+
+  template <int dim>
+  class Maxwell : public ParameterAcceptor
+  {
+  public:
+    Maxwell();
+    void run();
+
+  private:
+    /* run time parameters */
+    double       scaling;
+    unsigned int refinements;
+    unsigned int fe_order;
+    unsigned int quadrature_order;
+    unsigned int n_outputs;
+
+    void parse_parameters_callback();
+    void make_grid();
+    void setup_system();
+    void assemble_system();
+    void solve();
+    void output_results();
+
+    Parameters<dim>            parameters;
+    PerfectlyMatchedLayer<dim> perfectly_matched_layer;
+
+    Triangulation<dim> triangulation;
+    DoFHandler<dim>    dof_handler;
+
+    std::unique_ptr<FiniteElement<dim>> fe;
+
+    AffineConstraints<double> constraints;
+    SparsityPattern           sparsity_pattern;
+    SparseMatrix<double>      system_matrix;
+    Vector<double>            solution;
+    Vector<double>            system_rhs;
+  };
+
+
+  // @sect4{The Constructor}
+  // The Constructor simply consists specifications for the mesh
+  // and the order of the fnite elements. These are editable through
+  // the .prm file.
+
+  template <int dim>
+  Maxwell<dim>::Maxwell()
+    : ParameterAcceptor("Maxwell")
+    , dof_handler(triangulation)
+  {
+    ParameterAcceptor::parse_parameters_call_back.connect(
+      std::bind(&Maxwell<dim>::parse_parameters_callback, this));
+
+    scaling = 20;
+    add_parameter("scaling", scaling, "scale of the hypercube geometry");
+
+    refinements = 6;
+    add_parameter("refinements",
+                  refinements,
+                  "number of refinements of the geometry");
+
+    fe_order = 0;
```

Comment:
Is there a specific reason for the choice of linear Nédélec elements?

I would suggest to use `fe_order=1;` (and therefore maybe reduce the number of refinements). 

Because if we consider a constant number of DoFs, the result will look much better using fe_order=1, since the higher polynomial Nédélec elements can better approximate the wave characteristics of the solution.

Reply:
@kinnewig We have added runtime parameters to control the order of the element and quadrature. Thanks!